### PR TITLE
feat: add editChannelOverwrite()

### PR DESCRIPTION
**Pull request by itohatweb**
**Please describe the changes this PR makes and why it should be merged:**
Editing a Channel Permission is not intuitive with editChannel. editChannelOverwrite function was requested by Skillz4Killz .
Should this function take an array of objects instead of one?

**Status**

- [x] Code changes have been tested against the Discord API, or there are no code changes

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/handlers/channel.ts [hunk 1]
```diff
@@ -422,6 +424,55 @@ export async function editChannel(
...
+  }
+
+  const { allow, deny, ...info } = overwrite;
+  let channel = cache.channels.get(channelID);
+
+  if (!channel) return Errors.CHANNEL_NOT_FOUND;
+
+  const payload = {
+    permission_overwrites: [
+      ...(channel?.permissionOverwrites || []).map((ro) => ({
```
Comment (ayntee) [inline src/handlers/channel.ts:447]: `ro` is not-so explanatory, maybe rename it to `rawOverwrite`?

Comment (itohatweb) [reply src/handlers/channel.ts]: Done

File: src/handlers/channel.ts [hunk 2]
```diff
@@ -422,6 +424,55 @@ export async function editChannel(
...
+    ["MANAGE_CHANNELS"],
+  );
+  if (!hasManageChannelPerm) {
+    throw new Error(Errors.MISSING_MANAGE_CHANNELS);
+  }
+
+  const { allow, deny, ...info } = overwrite;
+  let channel = cache.channels.get(channelID);
+
+  if (!channel) return Errors.CHANNEL_NOT_FOUND;
```
Comment (ayntee) [inline src/handlers/channel.ts:443]:
```suggestion
  if (!channel) throw new Error(Errors.CHANNEL_NOT_FOUND);
```

**Comment by ayntee** [comment]
Is this tested?

**Comment by itohatweb** [comment]
Whole code was tested, exept that with the throw Error.

**Comment by Skillz4Killz** [comment]
I left a lot thoughts comments about this in your other PR. Please use those to apply them here and then resolve the conversation there. I didn't see this PR and just was reviewing the other one first

**Pull request merged**